### PR TITLE
Minimum sum partition using DP in C++

### PR DESCRIPTION
**Pull request by Meghag5**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes

### Description

(Write your answer here.)
This problem is solved using dp.
In this problem we have to find two subsets which will return minimum difference.
Input -
4
1 6 5 11
Output - 
1

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#5196

### CODE REVIEW

File: C-Plus-Plus/dp/Minimum_sum_partition.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*
+This problem is solved using dp.
+In this problem we have to find two subsets which will return minimum difference
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_sum_partition.cpp]: Remove this redundant description and add a proper description

File: C-Plus-Plus/dp/Minimum_sum_partition.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        
+       
+       ans = minimumSumPartition(arr, sum/2, N);
+
+       if (sum & 1)
+            ans++;
+
+        cout << ans << "\n";
+       
+    }
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_sum_partition.cpp]: Remove test case part as we are not using online judge use `cout` for proper display of input and Output.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Meghag5** [comment]
pr updated @ankitaggarwal23 pls review

File: C-Plus-Plus/dp/Minimum_sum_partition.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
+/*
+This problem is solved using dp.
+In this problem we have to find minimum difference of two subset.
+We have to divide array in two subsets in such a way that difference of both subsets 
+will become minimum.
+*/
+#include <iostream>
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_sum_partition.cpp]: Replace header with `#include <bits/stdc++.h>`

File: C-Plus-Plus/dp/Minimum_sum_partition.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    for (int i = 1; i <= N; i++)
+    {
+        for (int j = 1; j <= sum; j++)
+        {
+            // if subset sum value less than sum then we will proceed further and we add value to it or we will leave it
+            if (arr[i - 1] <= j)
+                DP[i][j] = DP[i-1][j-arr[i-1]] || DP[i-1][j];
+            else // if subset sum value grater than sum then we will proceed further
+                DP[i][j] = DP[i-1][j];
+        }
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_sum_partition.cpp]: Remove redundant comments

File: C-Plus-Plus/dp/Minimum_sum_partition.cpp [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }
+    
+    //i will check in last row from right where i get first true subset sum....it will give my answer
+
+    int i = sum;
+
+    // till value of matrix of last row do not return true we will decrese value of i by 1
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_sum_partition.cpp]: Remove redundant comments and remove space and aligned it.

File: C-Plus-Plus/dp/Minimum_sum_partition.cpp [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+       
+       ans = minimumSumPartition(arr, sum/2, N);
+
+       if (sum & 1)
+            ans++;
+
+        cout << ans << "\n";
+
+    return 0;
+}
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_sum_partition.cpp]: Rewrite it like that,

```
int main()
{
    int N, ans;

    cout << "Enter the value of N: \n";
    cin >> N;
    int arr[N], sum = 0;

    //We will find the total sum of array to find the range of the array
    cout << "Enter the value of array: \n";
    for (int i = 0; i < N; i++)
    {
        cin >> arr[i];
        sum += arr[i];
    }


   ans = minimumSumPartition(arr, sum/2, N);

   if (sum & 1)
        ans++;

   cout << ans << "\n";

   return 0;
}
```

File: C-Plus-Plus/dp/Minimum_sum_partition.cpp [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+/*
+Input -
+4
+1 6 5 11
+Output - 
+1
+Time Complexity - O(N*sum)
+Space Complexity - O(N*sum)
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_sum_partition.cpp]: Rewrite it as

```
/*
Sample Output:

Input:
Enter the value of N:
4
Enter the value of array:
1 6 5 11

Output: 
1

Time Complexity - O(N*sum)
Space Complexity - O(N*sum)

*/
```

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Meghag5** [comment]
all changes made @ankitaggarwal23 pls review

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by gat786** [comment]
@ankitaggarwal23 this same codefile already exists in our repository. So, if this pr aims to improve the same file it should edit the same file and not create another file with same file name. I think after merging this PR there seems to be a problem on working with windows systems since their filenames are not case-sensitive

**Pull request merged**